### PR TITLE
Clean up referral code, promo code and deprecated methods

### PR DESCRIPTION
**Pull request by aaustin**
@ahmednawar @derrickstaten 

The following migrations will need to be posted in the repo:
- setDebug must be called on getInstance. It's no longer a static method.
- referral code and promo code methods are no longer available
- action count methods are no longer available
- getReferralUrl removed - please migrate to BUO
- getContentUrl removed - please migrate to BUO
- BUO showShareSheet andCallback method deprecated - migrate to showShareSheet callback.

### CODE REVIEW

File: Branch-SDK/Branch-SDK/BNCPreferenceHelper.m [hunk 1]
```diff
@@ -33,17 +33,12 @@
 NSString * const BRANCH_PREFS_KEY_SESSION_PARAMS = @"bnc_session_params";
 NSString * const BRANCH_PREFS_KEY_INSTALL_PARAMS = @"bnc_install_params";
 NSString * const BRANCH_PREFS_KEY_USER_URL = @"bnc_user_url";
```
Comment (ahmednawar) [inline Branch-SDK/Branch-SDK/BNCPreferenceHelper.m]: I don't see this key used anywhere as well

File: Branch-SDK/Branch-SDK/Branch.m [hunk 2]
```diff
@@ -325,8 +297,6 @@ - (void)initSessionWithLaunchOptions:(NSDictionary *)options isReferrable:(BOOL)
 
 - (void)initSessionWithLaunchOptions:(NSDictionary *)options isReferrable:(BOOL)isReferrable explicitlyRequestedReferrable:(BOOL)explicitlyRequestedReferrable automaticallyDisplayController:(BOOL)automaticallyDisplayController {
     self.shouldAutomaticallyDeepLink = automaticallyDisplayController;
-    self.preferenceHelper.isReferrable = isReferrable;
```
Comment (ahmednawar) [inline Branch-SDK/Branch-SDK/Branch.m]: Shouldn't we change the method signature to exclude both arguments?

File: ChangeLog.md [hunk 3]
```diff
@@ -9,15 +9,6 @@ Branch iOS SDK change log
   * Simplify Carthage project
   * Fix framework headers visibility issue with BranchView
 
-- v0.12.3
```
Comment (ahmednawar) [inline ChangeLog.md]: Thanks for catching this!

**Comment by ahmednawar** [comment]
insanely satisfying PR 👍

File: Branch-SDK/Branch-SDK/BNCPreferenceHelper.m (hunk 1 above)
Comment (aaustin) [inline Branch-SDK/Branch-SDK/BNCPreferenceHelper.m]: it's used to store the user URL - we can't get rid of it

File: Branch-SDK/Branch-SDK/Branch.m (hunk 2 above)
Comment (aaustin) [inline Branch-SDK/Branch-SDK/Branch.m]: this was one of those changes I felt uncomfortable making as I think we had this method in the onboarding documentation for a long time.

File: Branch-SDK/Branch-SDK/BNCPreferenceHelper.m (hunk 1 above)
Comment (ahmednawar) [inline Branch-SDK/Branch-SDK/BNCPreferenceHelper.m]: ah, right. I was searching for `bnc_user_url` instead of `BRANCH_PREFS_KEY_USER_URL` :facepalm

**Pull request merged**